src/sysl/importers/import_swagger.py

```diff
@@ -94,200 +76,287 @@ def javaParam(match):
+            vocabulary_factory = load_vocabulary
+        self._logger = logger
+        self._param_cache = {}
+        self._words = set()
+        self._vocabulary_factory = vocabulary_factory
+
+    def warn(self, msg):
+        self._logger.warn(msg)
+
+    def words(self):
+        """Lazy-load WORDS."""
+        if not self._words:
+            self._words.update(self._vocabulary_factory())
+            if not self._words:
+                self.warn("could not load any vocabulary, janky environment-specific heuristics for renaming path template names may fail")
+        return self._words
+
+    def javaParam(self, match):
+        # TODO(anz-rfc) this seems janky and fragile.
+        param = match.group(1)
+        ident = self._param_cache.get(param)
+
+        if ident is None:
+            # foo-bar to fooBar
+            ident = re.sub(r'-(\w?)', lambda m: m.group(1).upper(), param)
+
+            # {fooid} -> fooId (only if foo is in WORDS but fooid isn't)
+            m = re.match(r'{([a-z]+)id}$', ident)
+            if m:
+                word = m.group(1)
+                if word in self.words() and word + 'id' not in self.words():
+                    ident = '{' + word + 'Id}'
+
+            ident = ident.replace('}', '<:string}')
+
+            self._param_cache[param] = ident
+
+        return ident
+
+    def translate_path_template_params(self, path):
+        # OAS 2.0 supports path templating.
+        # ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#pathTemplating
+        # For some reason, we rename these params via javaParam
+        return re.sub(r'({[^/]*?})', self.javaParam, path)
+
+    def translate(self, swag, appname, package, w):
+
+        if 'info' in swag:
+            def info_attrs(info, prefix=''):
+                for (name, value) in sorted(info.iteritems()):
+                    if isinstance(value, dict):
+                        info_attrs(value, prefix + name + '.')
+                    else:
+                        w('@{}{} = {}', prefix, name, json.dumps(value))
+
+            title = swag['info'].pop('title', '')
+            info_attrs(swag['info'])
+        else:
+            title = ''
+
+        if 'host' in swag:
+            w('@host = {}', json.dumps(swag['host']))
+
+        w(u'{}{} [package={}]:',
+            appname, title and ' ' + json.dumps(title), json.dumps(package))
+
+        with w.indent():
+            w(u'| {}', swag['info'].get('description', 'No description.'))
+
+            for (path, api) in sorted(swag['paths'].iteritems()):
+                w(u'\n{}:', self.translate_path_template_params(path))
+                with w.indent():
+                    if 'parameters' in api:
+                        del api['parameters']
+                    for (i, (method, body)) in enumerate(sorted(api.iteritems(),
+                                                                key=lambda t: METHOD_ORDER[t[0]])):
```

Comment:
opinion: I'm not a fan of mega-indents. I'd prefer a one-liner here or precompute `sorted(…)` before the loop.